src/helpers/utm-params.ts

```diff
@@ -0,0 +1,21 @@
+export const autoParseUTMParams = () => {
+  const params = new URLSearchParams(window.location.search);
+
+  const possible_params = [
+    "utm_source",
+    "utm_medium",
+    "utm_campaign",
+    "utm_term",
+    "utm_content",
+  ];
+  const utmParams: { [key: string]: string } = {};
+
+  possible_params.forEach((param) => {
+    const paramValue = params.get(param);
+    if (paramValue !== null) {
+      utmParams[param] = paramValue;
```

Comment:
I wonder if we should set these params to have a common parent, `utm`, as we do in [the behavioral events](https://github.com/RevenueCat/purchases-js/blob/144adca71a6039d05ac66c5d72d4f5df634814ec/src/behavioural-events/event-context.ts#L16-L22)

Reply:
+1

Reply:
RE: Common parent

@vicfergar I suspect you mean
```
metadata = {
  'someKey' : 'someValue',
  
  'utm': {
     'utm_source':'...',
     'utm_...':'...'
  }
}
```

I wouldn't add that nesting, considering that this will reach 2 places:
1. Stripe metadata (flat)
2. Our metadata/events

Adding a nested object would complicate all the rest down the line and I don't think it's worth it.

Reply:
RE: Sharing the code

I agree @polmiro, I've seen that you already reused the function in one of your PRs.
I'll keep you posted if I merge first.

Reply:
Oh, I noticed that the `purchase` endpoint does not accept nested attributes, right?

Reply:
Exactly, they should be flat so that they can be sent along to Stripe as metadata